ci/scripts/utils/launch_java_agent.sh

```diff
@@ -1,8 +1,37 @@
-#!/bin/env bash
+#!/bin/bash
+
+# ==============================================================================
+# Script Name: launch_java_agent.sh
+#
+# Description: This script automates the process of launching a Jenkins agent
+#              on a specified machine. It ensures that the necessary
+#              prerequisites are met, such as the availability of JAVA_HOME,
+#              the Jenkins agent launch directory, and proper authentication
+#              with GitHub.
+#
+#              It then proceeds to check if the Jenkins node is online and
+#              decides whether to launch the Jenkins agent based on the node's
+#              status. The agent is launched in the background,
+#              and its PID is logged for reference.
+#
+# Prerequisites: JAVA_HOME must be set to a valid JDK installation.
+#                GitHub CLI (gh) must be installed and authenticated.
```

Comment:
Perhaps a note (or link to a gist) on how to authenticate would be useful.

Also, please mention what version(s) of `gh` are required.